File: XMLCheck/Handler/CheckRulesXML09/Xml9RuleRunner.cs
using XMLCheck.Handler.CheckRulesXML08;
using XMLCheck.Models;
using XMLCheck.Services;

namespace XMLCheck.Handler.CheckRulesXML09
{
    public static class Xml9RuleRunner
    {
        private static readonly List<IXml9Rule> _rules = new()
        {

        };
        public static List<ErrorDetails> Run(List<Xml9Model>? models,HoSoContext context)
        {
            var errors = new List<ErrorDetails>();

            if (models == null || models.Count == 0)
                return errors;

            foreach (var model in models)
            {
                foreach (var rule in _rules)
                {
                    var err = rule.Check(model,context);
                    if (err != null)
                        errors.Add(err);
                }
            }

            return errors;
        }
    }
}

File: XMLValidator_v2/src/parsers/rules/xml9/xml9RuleRunner.ts
import { Xml9Model } from '../../../interfaces/xml-models.interface';
import { HoSoContext } from '../../../services/hoso-context';
import { ErrorDetails } from '../../../dto/xml-check.dto';
import { IXml9Rule } from './xml9RuleBase';

export class Xml9RuleRunner {
  private static readonly rules: IXml9Rule[] = [];

  public static run(models: Xml9Model[], context: HoSoContext): ErrorDetails[] {
    const errors: ErrorDetails[] = [];
    if (!models || models.length === 0) return errors;

    for (const model of models) {
      for (const rule of this.rules) {
        const err = rule.check(model, context);
        if (err) {
          errors.push(err);
        }
      }
    }

    return errors;
  }
}
